mrtt-ui/src/components/PreRestorationAssessmentForm.js

```diff
@@ -0,0 +1,520 @@
+import { useState, useCallback } from 'react'
```

Comment:
I needed to add a save button, if it's an aesthetic thing, I can change the kind it is . I had to add it due to the way `useFieldsArray` works:

If I save to add a measurement item to new row, this updates in the` TabularInputItem `component. Previously, it was using a controller, with the name referencing that specific index item. This was cuaing object labels to get added twice, for example we would get `{measurementType: xxxx, measurementValue: 23, measurementValue: 23}`. This was also happening when I was adding those initial measurement values, before any new rows were added. 

I realized to solve this, I had to be consistent using `useFieldArray` methods, vs updating a controller at the exact time I was updating the field array. I'm happy to explain this better through a call if you like. 

The save button allows us to update the `fieldArray` without creating a race condition in a controller. This may still be a less than ideal solution however, since the save button updates the current fieldsArray, but doesn't actually submit. So if you refresh, it will not be saved unless you actually submit the form.

two major things I think we need to add to create a better user experience is:

1. Show all validation errors by the submit button, in addition to the actual question
2. Have some sort of indicator that the entire form is not saved/submitted once any edit is made

@alanjleonard @mmnoo 

That being said, I will also add the unit row item. Hopefully this makes sense, feel free to ping me if you feel you want to go over it more

Reply:
as for 5.3f, this is intentional. @mmnoo 

If no countries are selected in section 1, no mangrove species will appear in 5.3e,

If no mangrove species are selected in 5.3e, 5.3f will not appear 

![Uploading Screen Shot 2022-07-06 at 12.10.28 PM.png…]()

Reply:
Not sure about 5.3f. What I was talking about is the input _was_ showing so I put a value in, and submitted it. It didnt render in the input on page refresh.